controllers/backstage_controller.go

```diff
@@ -147,144 +133,46 @@ func (r *BackstageReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
+		}
 	}
 	return nil
 }
 
-func readYamlFile(path string, object interface{}) error {
-
-	b, err := os.ReadFile(path)
-	if err != nil {
-		return fmt.Errorf("failed to read YAML file: %w", err)
-	}
-	return readYaml(b, object)
-}
-
-func defFile(key string) string {
-	return filepath.Join(os.Getenv("LOCALBIN"), "default-config", key)
-}
-
-// sets the RuntimeRunning condition
-func (r *BackstageReconciler) setRunningStatus(ctx context.Context, backstage *bs.Backstage, ns string) {
-
-	meta.SetStatusCondition(&backstage.Status.Conditions, v1.Condition{
-		Type:               bs.RuntimeConditionRunning,
-		Status:             "Unknown",
-		LastTransitionTime: v1.Time{},
-		Reason:             "Unknown",
-		Message:            "Runtime in unknown status",
-	})
-}
-
-// sets the RuntimeSyncedWithConfig condition
-func (r *BackstageReconciler) setSyncStatus(backstage *bs.Backstage) {
-
-	status := v1.ConditionUnknown
-	reason := "Unknown"
-	message := "Sync in unknown status"
-	if r.OwnsRuntime {
-		status = v1.ConditionTrue
-		reason = "Synced"
-		message = "Backstage syncs runtime"
-	}
-
-	meta.SetStatusCondition(&backstage.Status.Conditions, v1.Condition{
-		Type:               bs.RuntimeConditionSynced,
-		Status:             status,
-		LastTransitionTime: v1.Time{},
-		Reason:             reason,
-		Message:            message,
-	})
-}
-
-// sets backstage-{Id} for labels and selectors
-func setBackstageAppLabel(labels *map[string]string, backstage bs.Backstage) {
-	if *labels == nil {
-		*labels = map[string]string{}
-	}
-	(*labels)[BackstageAppLabel] = fmt.Sprintf("backstage-%s", backstage.Name)
-}
-
-// sets backstage-psql-{Id} for labels and selectors
-func setBackstageLocalDbLabel(labels *map[string]string, name string) {
-	if *labels == nil {
-		*labels = map[string]string{}
-	}
-	(*labels)[BackstageAppLabel] = name
-}
-
-// sets labels on Backstage's instance resources
-func (r *BackstageReconciler) labels(meta *v1.ObjectMeta, backstage bs.Backstage) {
-	if meta.Labels == nil {
-		meta.Labels = map[string]string{}
-	}
-	meta.Labels["app.kubernetes.io/name"] = "backstage"
-	meta.Labels["app.kubernetes.io/instance"] = backstage.Name
-	//meta.Labels[BackstageAppLabel] = fmt.Sprintf("backstage-%s", backstage.Name)
-}
-
 // SetupWithManager sets up the controller with the Manager.
 func (r *BackstageReconciler) SetupWithManager(mgr ctrl.Manager, log logr.Logger) error {
 
-	if len(r.PsqlImage) == 0 {
-		r.PsqlImage = "quay.io/fedora/postgresql-15:latest"
-		log.Info("Enviroment variable is not set, default is used", bs.EnvPostGresImage, r.PsqlImage)
-	}
-
-	if len(r.BackstageImage) == 0 {
-		r.BackstageImage = "quay.io/janus-idp/backstage-showcase:next"
-		log.Info("Enviroment variable is not set, default is used", bs.EnvBackstageImage, r.BackstageImage)
-	}
-
 	builder := ctrl.NewControllerManagedBy(mgr).
 		For(&bs.Backstage{})
 
 	if r.OwnsRuntime {
 		builder.Owns(&appsv1.Deployment{}).
```

Comment:
My concern is that this will cause the operator to watch ALL deployment, service and statefulset objects in the entire cluster. This can lead to dramatic resource consumptions in a production cluster where there are large number of such clusters. I think we should drop this until we are able to do a load/performance test to evaluate the performance impact.

Reply:
OK for me, it was kinda experimental thing not related to this PR (you can see something like this in the main branch too ;) )